components/renderers/controls/array/array-renderer.ts

```diff
@@ -2,210 +2,134 @@
+        let items = resolvedSubSchema.items;
+        let label = element.label ? element.label : "";
+
+        // TODO: think about how to access options in an uniform fashion
+        if (element.options != undefined && element.options['simple']) {
+            let controlDescription = JSONForms.RenderDescriptionFactory.createControlDescription(schemaPath, services, element);
+            let properties = _.keys(items['properties']);
+            let propertiesString = JSON.stringify(properties);
+            controlDescription.template = `<jsonforms-layout class="jsf-group">
+              <fieldset>
+                <legend>${label}</legend>
+                <div ng-repeat='d in element.instance[element.path]'>
+                    <div ng-repeat='prop in ${propertiesString}'>
+                    <strong>{{prop | capitalize}}:</strong> {{d[prop]}}
+                    </div>
+                    <hr ng-show="!$last">
+                </div>
+               </fieldset>
+             </jsonforms-layout>`;
+            return controlDescription;
         } else {
-            // TODO is this case even possible (array containing primitive type)?
+            let submitElement = {};
+            let supportsSubmit = element.options != undefined && element.options['submit'];
+            let generatedGroups = this.createControlGroupPerItem(schemaPath, items, data.length);
+            let buttonText = JSONForms.PathUtil.beautifiedLastFragment(schemaPath);
+
+            let template = `
+            <jsonforms-layout class="jsf-group">
+              <fieldset>
+                <legend>${label}</legend>
+                <div ng-repeat="renderDescription in element.generateControlDescriptions(element.instance[element.path]) ">
+                  <jsonforms-dynamic-widget element="renderDescription">
+                  </jsonforms-dynamic-widget>
+                </div>
+                  <jsonforms-dynamic-widget ng-repeat="submitRenderDescription in element.submitControls" element="submitRenderDescription">
+                  </jsonforms-dynamic-widget>
+               </fieldset>
+               <input ng-show="${supportsSubmit}" type="button" value="Add to ${buttonText}" ng-click="element.submitCallback()" ng-model="element.submitElement">
+               </input>
+             </jsonforms-layout>`;
+
+            let containeeDescriptions = JSONForms.RenderDescriptionFactory.renderElements(generatedGroups, this.renderService, services);
+            let containerDescription = JSONForms.RenderDescriptionFactory.createContainerDescription(this.maxSize, containeeDescriptions, template, services, element);
+            containerDescription['submitElement'] = submitElement;
+            containerDescription['submitControls'] = this.createControlsForSubmit(items, schemaPath, submitElement, services);
+            containerDescription['submitCallback'] = () => data.push(_.clone(submitElement));
```

Comment:
is data not the whole object? so must we not first find the correct inner element using the schemapath?